perl test fails during build

@rfugina thank you for adding the license information. When I went to build the latest development branch, the perl test failed. Here are the results of the test:
```
Executing(%check): /bin/sh -e /var/tmp/rpm-tmp.fIvOtU
+ umask 022
+ cd /home/abauer/rpmbuild/BUILD
+ cd Astro-SunTime-0.01
+ /usr/bin/make -O -j2 test
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00load.t .... ok
# Test 3 got: "7:35" (t/01detail.t at line 13)
#   Expected: "6:35"
#  t/01detail.t line 13 is: ok(sun_time(type => 'rise', latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 1970'), '6:35');
# Test 4 got: "20:40" (t/01detail.t at line 14)
#   Expected: "19:40"
#  t/01detail.t line 14 is: ok(sun_time(type => 'set',  latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 1970'), '19:40');
# Test 7 got: "7:36" (t/01detail.t at line 17)
#   Expected: "6:36"
#  t/01detail.t line 17 is: ok(sun_time(type => 'rise', latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 2020'), '6:36');
# Test 8 got: "20:38" (t/01detail.t at line 18)
#   Expected: "19:38"
#  t/01detail.t line 18 is: ok(sun_time(type => 'set',  latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 2020'), '19:38');
# Test 11 got: "8:10" (t/01detail.t at line 22)
#    Expected: "7:10"
#  t/01detail.t line 22 is: ok(sun_time(type => 'rise', latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 1970'), '7:10');
# Test 12 got: "19:11" (t/01detail.t at line 23)
#    Expected: "18:11"
#  t/01detail.t line 23 is: ok(sun_time(type => 'set',  latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 1970'), '18:11');
# Test 15 got: "8:08" (t/01detail.t at line 26)
#    Expected: "7:08"
#  t/01detail.t line 26 is: ok(sun_time(type => 'rise', latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 2020'), '7:08');
# Test 16 got: "19:12" (t/01detail.t at line 27)
#    Expected: "18:12"
#  t/01detail.t line 27 is: ok(sun_time(type => 'set',  latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 2020'), '18:12');
t/01detail.t .. 
Failed 8/16 subtests 

Test Summary Report
-------------------
t/01detail.t (Wstat: 0 Tests: 16 Failed: 8)
  Failed tests:  3-4, 7-8, 11-12, 15-16
Files=2, Tests=17,  0 wallclock secs ( 0.01 usr  0.02 sys +  0.04 cusr  0.01 csys =  0.08 CPU)
Result: FAIL
Failed 1/2 test programs. 8/17 subtests failed.
Makefile:812: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
error: Bad exit status from /var/tmp/rpm-tmp.fIvOtU (%check)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.fIvOtU (%check)
```
Since the values appear to be off by an hour, I'm guessing commit 63e070d749303031f7dc88f7962363d590f7c204 introduced this.

Some additional data to help isolate the problem.

Using traditional make instead of rpmbuild or mock, the following was performed on Fedora 25:
```
[abauer@vFedora25 specfiles]$ ls -l /etc/localtime
lrwxrwxrwx. 1 root root 37 Aug 17  2013 /etc/localtime -> ../usr/share/zoneinfo/America/Chicago

[abauer@vFedora25 Astro-SunTime-0.01]$ perl Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Astro::SunTime
Writing MYMETA.yml and MYMETA.json

[abauer@vFedora25 Astro-SunTime-0.01]$ make test
cp SunTime.pm blib/lib/Astro/SunTime.pm
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00load.t .... ok   
t/01detail.t .. 1/16 # Test 3 got: "7:35" (t/01detail.t at line 13)
#   Expected: "6:35"
#  t/01detail.t line 13 is: ok(sun_time(type => 'rise', latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 1970'), '6:35');
# Test 4 got: "20:40" (t/01detail.t at line 14)
#   Expected: "19:40"
#  t/01detail.t line 14 is: ok(sun_time(type => 'set',  latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 1970'), '19:40');
# Test 7 got: "7:36" (t/01detail.t at line 17)
#   Expected: "6:36"
#  t/01detail.t line 17 is: ok(sun_time(type => 'rise', latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 2020'), '6:36');
# Test 8 got: "20:38" (t/01detail.t at line 18)
#   Expected: "19:38"
#  t/01detail.t line 18 is: ok(sun_time(type => 'set',  latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 2020'), '19:38');
# Test 11 got: "8:10" (t/01detail.t at line 22)
#    Expected: "7:10"
#  t/01detail.t line 22 is: ok(sun_time(type => 'rise', latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 1970'), '7:10');
# Test 12 got: "19:11" (t/01detail.t at line 23)
#    Expected: "18:11"
#  t/01detail.t line 23 is: ok(sun_time(type => 'set',  latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 1970'), '18:11');
# Test 15 got: "8:08" (t/01detail.t at line 26)
#    Expected: "7:08"
#  t/01detail.t line 26 is: ok(sun_time(type => 'rise', latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 2020'), '7:08');
# Test 16 got: "19:12" (t/01detail.t at line 27)
#    Expected: "18:12"
#  t/01detail.t line 27 is: ok(sun_time(type => 'set',  latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 2020'), '18:12');
t/01detail.t .. Failed 8/16 subtests 

Test Summary Report
-------------------
t/01detail.t (Wstat: 0 Tests: 16 Failed: 8)
  Failed tests:  3-4, 7-8, 11-12, 15-16
Files=2, Tests=17,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.05 cusr  0.01 csys =  0.10 CPU)
Result: FAIL
Failed 1/2 test programs. 8/17 subtests failed.
Makefile:839: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
```

Same test, but performed on Ubuntu 16.10:
```
abauer@Ubuntu-Desktop:~/git/perl-astro-suntime$ ls -l /etc/localtime
lrwxrwxrwx 1 root root 35 Jan  6 20:03 /etc/localtime -> /usr/share/zoneinfo/America/Chicago

abauer@Ubuntu-Desktop:~/git/perl-astro-suntime$ perl Makefile.PL
Generating a Unix-style Makefile
Writing Makefile for Astro::SunTime
Writing MYMETA.yml and MYMETA.json

abauer@Ubuntu-Desktop:~/git/perl-astro-suntime$ make test
cp SunTime.pm blib/lib/Astro/SunTime.pm
PERL_DL_NONLAZY=1 PERL_USE_UNSAFE_INC=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00load.t .... ok   
t/01detail.t .. 1/16 # Test 3 got: "7:35" (t/01detail.t at line 13)
#   Expected: "6:35"
#  t/01detail.t line 13 is: ok(sun_time(type => 'rise', latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 1970'), '6:35');
# Test 4 got: "20:40" (t/01detail.t at line 14)
#   Expected: "19:40"
#  t/01detail.t line 14 is: ok(sun_time(type => 'set',  latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 1970'), '19:40');
# Test 7 got: "7:36" (t/01detail.t at line 17)
#   Expected: "6:36"
#  t/01detail.t line 17 is: ok(sun_time(type => 'rise', latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 2020'), '6:36');
# Test 8 got: "20:38" (t/01detail.t at line 18)
#   Expected: "19:38"
#  t/01detail.t line 18 is: ok(sun_time(type => 'set',  latitude => 40, longitude => -77, time_zone => -4, date => '1 Sep 2020'), '19:38');
# Test 11 got: "8:10" (t/01detail.t at line 22)
#    Expected: "7:10"
#  t/01detail.t line 22 is: ok(sun_time(type => 'rise', latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 1970'), '7:10');
# Test 12 got: "19:11" (t/01detail.t at line 23)
#    Expected: "18:11"
#  t/01detail.t line 23 is: ok(sun_time(type => 'set',  latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 1970'), '18:11');
# Test 15 got: "8:08" (t/01detail.t at line 26)
#    Expected: "7:08"
#  t/01detail.t line 26 is: ok(sun_time(type => 'rise', latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 2020'), '7:08');
# Test 16 got: "19:12" (t/01detail.t at line 27)
#    Expected: "18:12"
#  t/01detail.t line 27 is: ok(sun_time(type => 'set',  latitude => -45, longitude => 170, time_zone => 12, date => '1 Sep 2020'), '18:12');
t/01detail.t .. Failed 8/16 subtests 

Test Summary Report
-------------------
t/01detail.t (Wstat: 0 Tests: 16 Failed: 8)
  Failed tests:  3-4, 7-8, 11-12, 15-16
Files=2, Tests=17,  0 wallclock secs ( 0.02 usr  0.00 sys +  0.03 cusr  0.00 csys =  0.05 CPU)
Result: FAIL
Failed 1/2 test programs. 8/17 subtests failed.
Makefile:809: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
```
So it doesn't appear the issue has got to do with the build process or the target distro.... but I just now noticed that all the failed tests are during the month of September.

--- a/SunTime.pm
+++ b/SunTime.pm
@@ -116,7 +116,7 @@ sub sun_time
    $V = &normalize($V, $D);
    $V = $V * 3.81972;
 
-   my $hour = int($V) - $suntime[8]; # subtract value of daylight savings
+   my $hour = int($V);
    my $min  = int(($V - $hour) * 60 + 0.5);
 
    @suntime[2,1,0] = ($hour, $min, 0);

--- a/SunTime.pm
+++ b/SunTime.pm
@@ -116,10 +116,10 @@ sub sun_time
    $V = &normalize($V, $D);
    $V = $V * 3.81972;
 
-   my $hour = int($V);
+   my $hour = int($V) - $suntime[8]; # subtract value of daylight savings
    my $min  = int(($V - $hour) * 60 + 0.5);
 
-   @suntime[2,1,0,8] = ($hour, $min, 0, 0);
+   @suntime[2,1,0] = ($hour, $min, 0);
 
    @suntime = localtime(mktime(@suntime)); # normalize date structure
 

